Assets/Scripts/Boids/BoidSystem.cs

```diff
@@ -0,0 +1,385 @@
+            [DeallocateOnJobCompletion] [ReadOnly] public NativeArray<float>   cellObstacleDistance;
+            [DeallocateOnJobCompletion] [ReadOnly] public NativeArray<int>     cellTargetPositionIndex;
+            [DeallocateOnJobCompletion] [ReadOnly] public NativeArray<int>     cellCount;
+
+            public void Execute(Entity entity, int index, ref LocalToWorld localToWorld)
+            {
+                // temporarily storing the values for code readability
+                var forward                           = localToWorld.Forward;
+                var currentPosition                   = localToWorld.Position;
+                var cellIndex                         = cellIndices[index];
+                var neighborCount                     = cellCount[cellIndex];
+                var alignment                         = cellAlignment[cellIndex];
+                var separation                        = cellSeparation[cellIndex];
+                var nearestObstacleDistance           = cellObstacleDistance[cellIndex];
+                var nearestObstaclePositionIndex      = cellObstaclePositionIndex[cellIndex];
+                var nearestTargetPositionIndex        = cellTargetPositionIndex[cellIndex];
+                var nearestObstaclePosition           = obstaclePositions[nearestObstaclePositionIndex];
+                var nearestTargetPosition             = targetPositions[nearestTargetPositionIndex];
+
+                // steering calculations based on the boids algorithm
+                var obstacleSteering                  = currentPosition - nearestObstaclePosition;
+                var avoidObstacleHeading              = (nearestObstaclePosition + math.normalizesafe(obstacleSteering)
+                                                        * settings.ObstacleAversionDistance)- currentPosition;
+                var targetHeading                     = settings.TargetWeight
+                                                        * math.normalizesafe(nearestTargetPosition - currentPosition);
+                var nearestObstacleDistanceFromRadius = nearestObstacleDistance - settings.ObstacleAversionDistance;
+                var alignmentResult                   = settings.AlignmentWeight
+                                                        * math.normalizesafe((alignment/neighborCount)-forward);
+                var separationResult                  = settings.SeparationWeight
+                                                        * math.normalizesafe((currentPosition * neighborCount) - separation);
+                var normalHeading                     = math.normalizesafe(alignmentResult + separationResult + targetHeading);
+                var targetForward                     = math.select(normalHeading, avoidObstacleHeading, nearestObstacleDistanceFromRadius < 0);
+                var nextHeading                       = math.normalizesafe(forward + dt*(targetForward-forward));
```

Comment:
Consider changing these to use type types instead of var to make it easier to understand (unless var is needed for some other reason).

Reply:
Unresolved